sfputil/main.py

```diff
@@ -1280,6 +1335,10 @@ def upgrade(port_name, filepath):
 
     click.echo("Firmware run in mode 1 successful")
 
+    if is_fw_switch_done(port_name) != 1:
+        click.echo('Failed to switch firmware images!')
+        sys.exit(EXIT_FAIL)
+
```

Comment:
why are you checking for committed image inside this function when the commit_firmware() happens at line 1342